options-design/DESIGN.md

```diff
@@ -1,52 +1,123 @@
 # Options Design
-The design for options uses self referential functions and a builder in the mongo package for
-creating these. The main issue with the current structure is that it is difficult to discover which
-options are valid for what methods. This method also limits the way in which users can bundle
-together options that are passed through their application. The design laid out in this document
-attempts to solve these discoverability and bundling issues while ensuring that the useful self
-referential functions are kept intact. This design ensures that the core/options package is not
-exposed from the mongo package’s API, which has caused confusion for users in the past.
-
-This design uses packages for namespacing, and interfaces, types, and functions within those
-namespaces. Each namespace package contains an interface for each collection method that fall under
-the namespace, functions for each option that is available, and a bundle to facilitate creating
-groups of options. Since the bundle uses a fluent API for appending additional options, it can also
-be used for discovery of available options.
+The design for options uses [self referential
```

Comment:
If this is a design document for a feature, referencing the existing design isn't going to be helpful to future readers who have no knowledge of the previous design. I'd suggest you just highlight the benefits of the new design, and if you want to add a little bit about the previous design, I'd place it in a subsection somewhere towards the end.

Reply:
I think it's fine to leave it up top but change the title to Motivation (similar to server documents).  The way it is now I thought it was referring to the new design not the current design.

Reply:
I'll add this under the a Motivation heading and ensure to disambiguate the use of "this design".